tasks/src/test/java/webdriver/google_cloud/page/TenMinuteEmailHomePage.java

```diff
@@ -0,0 +1,62 @@
+package webdriver.google_cloud.page;
+
+import static webdriver.google_cloud.browser.Browser.waitForVisibleElement;
+
+import java.util.ArrayList;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class TenMinuteEmailHomePage extends GoogleCloudAbstractPage {
+  @FindBy(id = "mailAddress")
+  private WebElement mailInput;
+  @FindBy(xpath = "//*[@role='tab']")
+  private WebElement messagesList;
+  @FindBy(xpath = "//*[@id='mobilepadding']//h2")
+  private WebElement totalCostFromEmail;
+
+  private ArrayList<String> tabs;
+
+  public TenMinuteEmailHomePage(WebDriver driver) {
+    super(driver);
+  }
+
+  public String getTotalCostFromEmailText() {
+    return totalCostFromEmail.getText();
+  }
+
+  public void addNewTab() {
+    ((JavascriptExecutor)driver).executeScript("window.open()");
+  }
+
+  public void switchToTenMinutePageTab() {
+    tabs = new ArrayList<>(driver.getWindowHandles());
+    driver.switchTo().window(tabs.get(1));
+    driver.get("https://10minutemail.com");
+  }
+
+  public void switchToCalculatorPageTab() {
+    driver.switchTo().window(tabs.get(0));
+  }
+
+  public void copyAddress() {
```

Comment:
Such a complicated chain of actions more suitable to be stored in a service class. Page object should not be too smart, they only should provide a low-level abstraction of a logical page with atomic actions